### ISSUE TITLE
Default Service Ports Protocol Cannot be Added

### ISSUE THREAD

**Issue by dezzyfreeman** [quarkusio/quarkus#28297]
## Bug
The generated kubernetes.yml `Service` > `spec` > `ports` > `protocol` cannot be accessed using the following application.properties configuration:  `quarkus.kubernetes.ports."http".protocol=tcp`.

Using `quarkus.kubernetes.ports."http".container-port` and `quarkus.kubernetes.ports."http".host-port` as per the documentation, I am able to change the default kubernetes.yml `Service` > `spec` > `ports` > `targetPort` and the default kubernetes.yml `Service` > `spec` > `ports` > `port` respectively. I mention this to reference the syntax - I used the results of these latter configurations to confirm the syntax for the former is not the issue.

**application.properties**
```
quarkus.kubernetes.add-build-timestamp=false
quarkus.kubernetes.add-version-to-label-selectors=false
quarkus.kubernetes.prometheus.annotations=false

#  Append Ingress component.
quarkus.kubernetes.ingress.expose=true

#  metadata > name
quarkus.kubernetes.name=name

#  metadata > namespace
quarkus.kubernetes.namespace=namespace

quarkus.kubernetes.ports."http".protocol=tcp
quarkus.kubernetes.ports."http".container-port=8080
quarkus.kubernetes.ports."http".host-port=8080
```

**kubernetes.yml (service portion)**
```
apiVersion: v1
kind: Service
metadata:
  annotations:
    app.quarkus.io/commit-id: 5c4720d733cd2bd9bea27be621104d38e4609910
  labels:
    app.kubernetes.io/name: name
  name: name
  namespace: namespace
spec:
  ports:
    - name: http
      port: 8080
      targetPort: 8080
  selector:
    app.kubernetes.io/name: name
  type: ClusterIP
```

**Expected kubernetes.yml (service portion)**
```
apiVersion: v1
kind: Service
metadata:
  annotations:
    app.quarkus.io/commit-id: 5c4720d733cd2bd9bea27be621104d38e4609910
  labels:
    app.kubernetes.io/name: name
  name: name
  namespace: namespace
spec:
  ports:
    - name: http
      port: 8080
      targetPort: 8080
      protocol: TCP            # Here
  selector:
    app.kubernetes.io/name: name
  type: ClusterIP
```

## To Repoduce

1. Generate a Quarkus project
2. Add the extension quarkus-kubernetes
3. Add the application.properties file noted above
4. Run the project

**Quarkus Version**
```
<quarkus.platform.version>2.11.2.Final</quarkus.platform.version>
```

_On a side note:_
Is there any way to remove the default `annotations` completely? if possible I would  like to remove the `app.quarkus.io/commit-id` entriely, and replace the `app.kuberntes.io/name` key with just `name`. Couldn't find a way to achieve this in the documentation.

Thanks in advance.

**Comment by Sgitario** [comment]
There are several issues here:
- From one side, the property `quarkus.kubernetes.ports.<name>.protocol` is ignored. It will be fixed in: https://github.com/quarkusio/quarkus/pull/28310
- As you stated, the protocol property in the generated Service resource is not set. It needs to be fixed in Dekorate upstream. I've reported it: https://github.com/dekorateio/dekorate/issues/1068
- Finally, the only available protocols are TCP and UDP. However, we should support a few more (see the complete list [here](https://kubernetes.io/docs/concepts/services-networking/service/#application-protocol)). This needs to be done also in Dekorate.
- About removing the annotations entirely, this is not possible at the moment. If the extension detects you're using a repo, it will add the commit-id annotations.

**Comment by dezzyfreeman** [comment]
@Sgitario 

Thank you for your help. I am fairly new to Git Issues (this is my first one) so I have a clarifying question. Once a PR is merged into the main branch, is a release automatically generated? Or do the releases follow some arbitrary schedule where all merged pull requests since the last release are included?

If its the latter, is there somewhere I could find more information about when the next release including these PRs will be publicized?

**Comment by Sgitario** [comment]
Hi @dezzyfreeman , it's the latter. You can find more information about the release planning in https://github.com/quarkusio/quarkus/wiki/Release-Planning.
